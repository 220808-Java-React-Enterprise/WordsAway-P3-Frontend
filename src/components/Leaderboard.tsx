import axios, { Axios, AxiosResponse } from "axios"
import { useEffect, useState } from "react"
import { User } from "../types/User.type"
import WORDS_API from "../utils/ApiConfig"
import '../css/leaderboard.css'
export function getLeaders() {}
export function getRank() {}




const Leaderboard = () => {
    const [users, setUsers] = useState<User[]>([])
    const [user, setUser] = useState<User | null>(null);
    const [rank, setRank] = useState<number | null>(null);
    

    useEffect(()=>{
      const data = sessionStorage.getItem("user");
      if(data != null) setUser(JSON.parse(data));
      console.log(user);
    }, [])




     async function getLeaders() {
        await WORDS_API.get('/getTopTenElo')
        .then((response: AxiosResponse) => {
          console.log(response.data)
          setUsers(response.data)
        })
        .catch(() => (window.location.href = '/login'))
      }
      useEffect(() => {
        getLeaders()
        console.log(users);
      }, []);

      async function getRank(){
        await WORDS_API.get('/getRankElo')
        .then((response: AxiosResponse) =>{
          console.log(response.data)
          setRank(response.data)
        })
        .catch(()=>(window.location.href = '/login'))
      }
      useEffect(() =>{
        getRank()
        console.log("this is your rank "+ rank)
      })

 

      return (
        <div data-testid="leaderboard" className="leaderboard">
          <header>
            <h1>Leader Board</h1>
            <img src="assets\wizard3.png" alt="?"></img>
          </header>
          <table>
            <thead>
              <tr>
                <th className="rank">Ranking</th>
                <th className="username">Username</th>
                <th className="mmr">MMR</th>
                <th className="winstreak">Games Won</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td className="rank">1</td>
                <td className="username">{users[0]?.username}</td>
                <td className="mmr">{users[0]?.elo}</td>
                <td className="winstreak">{users[0]?.gamesWon}</td>
              </tr>
              <tr>
                <td className="rank">2</td>
                <td className="username">{users[1]?.username}</td>
                <td className="mmr">{users[1]?.elo}</td>
                <td className="winstreak">{users[1]?.gamesWon}</td>
              </tr>
              <tr>
                <td className="rank">3</td>
                <td className="username">{users[2]?.username}</td>
                <td className="mmr">{users[2]?.elo}</td>
                <td className="winstreak">{users[2]?.gamesWon}</td>
              </tr>
              <tr>
                <td className="rank">4</td>
                <td className="username">{users[3]?.username}</td>
                <td className="mmr">{users[3]?.elo}</td>
                <td className="winstreak">{users[3]?.gamesWon}</td>
              </tr>
              <tr>
                <td className="rank">5</td>
                <td className="username">{users[4]?.username}</td>
                <td className="mmr">{users[4]?.elo}</td>
                <td className="winstreak">{users[4]?.gamesWon}</td>
              </tr>
              <tr>
                <td className="rank">6</td>
                <td className="username">{users[5]?.username}</td>
                <td className="mmr">{users[5]?.elo}</td>
                <td className="winstreak">{users[5]?.gamesWon}</td>
              </tr>
              <tr>
                <td className="rank">7</td>
                <td className="username">{users[6]?.username}</td>
                <td className="mmr">{users[6]?.elo}</td>
                <td className="winstreak">{users[6]?.gamesWon}</td>
              </tr>
              <tr>
                <td className="rank">8</td>
                <td className="username">{users[7]?.username}</td>
                <td className="mmr">{users[7]?.elo}</td>
                <td className="winstreak">{users[7]?.gamesWon}</td>
                </tr>
                <tr>
                <td className="rank">9</td>
                <td className="username">{users[8]?.username}</td>
                <td className="mmr">{users[8]?.elo}</td>
                <td className="winstreak">{users[8]?.gamesWon}</td>
              </tr>
              <tr>
                <td className="rank">10</td>
                <td className="username">{users[9]?.username}</td>
                <td className="mmr">{users[9]?.elo}</td>
                <td className="winstreak">{users[9]?.gamesWon}</td>
                </tr>
                <tr>
                <td className="rank">{rank}</td>
                <td className="username">{user?.username}</td>
                <td className="mmr">{user?.elo}</td>
                <td className="winstreak">{user?.gamesWon}</td>
              </tr>
            </tbody>
          </table>
        </div>

    );
    }
    


export default Leaderboard

