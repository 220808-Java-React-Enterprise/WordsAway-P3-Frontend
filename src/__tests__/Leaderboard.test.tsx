import React from 'react'
import Leaderboard, {getLeaders, getRank} from '../components/Leaderboard'
import '@testing-library/jest-dom/extend-expect'
import ReactDOM from 'react-dom'
import { cleanup, render } from '@testing-library/react'
import renderer from "react-test-renderer"


    //for clearing overlapping tests
    afterEach(cleanup);
    jest.mock("../utils/ApiConfig")

    //syntax for checking if the div renders
    it('renders leaderboard without crashing', () => {
        const div = document.createElement("div");
        ReactDOM.render(<Leaderboard></Leaderboard>, div)
    })

    //syntax for checking labels inside tags
    it('renders leaderboard correctly', () => {
       const {getByTestId} = render(<Leaderboard/>)
       expect(getByTestId("leaderboard")).toBeEmptyDOMElement()
    })

    //syntax for checking if element jsonifies, creates a snapshot folder to check against
    //can save mulitple snapshots to check different props/labels
    it("matches snapshot", () => {
        const tree = renderer.create(<Leaderboard></Leaderboard>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should return an array of top ten', () =>{
        const result = getLeaders();
        console.log(result)
        expect(result).toBe([])
    })

    it('should return rank int', ()=>{
        const result = getRank();
        console.log(result)
        expect(result).toBe(Number)
    })