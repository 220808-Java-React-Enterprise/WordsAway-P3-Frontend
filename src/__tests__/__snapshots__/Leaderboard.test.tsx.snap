// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="leaderboard"
>
  <header>
    <h1>
      Leader Board
    </h1>
    <img
      alt="?"
      src="assets\\wizard3.png"
    />
  </header>
  <table>
    <thead>
      <tr>
        <th
          className="rank"
        >
          Ranking
        </th>
        <th
          className="username"
        >
          Username
        </th>
        <th
          className="mmr"
        >
          MMR
        </th>
        <th
          className="winstreak"
        >
          Games Won
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="rank"
        >
          1
        </td>
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
      <tr>
        <td
          className="rank"
        >
          2
        </td>
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
      <tr>
        <td
          className="rank"
        >
          3
        </td>
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
      <tr>
        <td
          className="rank"
        >
          4
        </td>
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
      <tr>
        <td
          className="rank"
        >
          5
        </td>
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
      <tr>
        <td
          className="rank"
        >
          6
        </td>
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
      <tr>
        <td
          className="rank"
        >
          7
        </td>
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
      <tr>
        <td
          className="rank"
        >
          8
        </td>
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
      <tr>
        <td
          className="rank"
        >
          9
        </td>
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
      <tr>
        <td
          className="rank"
        >
          10
        </td>
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
      <tr>
        <td
          className="rank"
        />
        <td
          className="username"
        />
        <td
          className="mmr"
        />
        <td
          className="winstreak"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
